src/ripple/app/consensus/LedgerConsensus.cpp

```diff
@@ -372,7 +374,8 @@ class LedgerConsensusImp
       @param map      the transaction set.
       @param acquired true if we have acquired the transaction set.
     */
-    void mapComplete (uint256 const& hash, SHAMap::ref map, bool acquired)
+    void mapComplete (uint256 const& hash, std::shared_ptr<SHAMap> const& map,
+                      bool acquired)
```

Comment:
Is this really an improvement? It's more verbose and more difficult to understand because you have to process the `shared_ptr` and the `const&`. It conflicts with the convention used throughout the rest of rippled that `ref` indicates a const reference to a shared pointer. It's also harder to audit because you can easily find instances of `::ref`, but finding instances of `std::shared_ptr ... const&` is not so easy.